### PR TITLE
Add better user-facing error messages

### PR DESCRIPTION
**Pull request by dbkr**
Fixes https://github.com/vector-im/vector-web/issues/1101

### CODE REVIEW

File: src/components/structures/TimelinePanel.js [hunk 1]
```diff
@@ -588,8 +588,8 @@ var TimelinePanel = React.createClass({
                 };
             }
             Modal.createDialog(ErrorDialog, {
-                title: "Failed to load event",
-                description: msg,
+                title: "Failed to load timeline position",
+                description: "Vector was trying to load a specific point in this room's timeline but was unable to find it.",
```
Comment (richvdh) [inline src/components/structures/TimelinePanel.js]: I think we probably need to distinguish between 404 and 403 here?

File: src/components/views/dialogs/NeedToRegisterDialog.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * Usage:
+ * Modal.createDialog(NeedToRegisterDialog, {
+ *   title: "some text", (default: "Registration required")
+ *   description: "some more text",
+ *   onClose: someFunction,
```
Comment (richvdh) [inline src/components/views/dialogs/NeedToRegisterDialog.js]: s/onClose/onFinished/

File: src/components/views/dialogs/NeedToRegisterDialog.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    {this.props.title}
+                </div>
+                <div className="mx_Dialog_content">
+                    {this.props.description}
+                </div>
+                <div className="mx_Dialog_buttons">
+                    <button onClick={this.onRegisterClicked}>
+                        Register
+                    </button>
+                    <button onClick={this.props.onFinished} autoFocus={this.props.focus}>
```
Comment (richvdh) [inline src/components/views/dialogs/NeedToRegisterDialog.js]: need to add `focus` to `propTypes`

File: src/components/structures/TimelinePanel.js (hunk 1 above)
Comment (dbkr) [inline src/components/structures/TimelinePanel.js]: As far as I can see, the context API never returns 403, although I might have missed somewhere. Would you expect that to be if you don't have permission to view the event in question?

**Comment by dbkr** [comment]
ptal

**Comment by richvdh** [comment]
lgtm

**Pull request merged**